Sources/ArcGISToolkit/FloorFilter/SiteFacilityPromptViewController.swift

```diff
@@ -0,0 +1,264 @@
+            
+        if (isShowingFacilities) {
+            if (indexPath.row <= facilities.count-1) {
+                cell.siteFacilityNameLabel.text = facilities[indexPath.row].name
+                cell.accessoryType = .none
+                    
+                // Highlight any previously selected Facility.
+                if (cell.siteFacilityNameLabel.text == viewModel.selectedFacility?.name) {
+                    cell.siteFacilityDotImg.isHidden = false
+                    cell.siteFacilityNameLabel?.font = UIFont.boldSystemFont(ofSize: 16.0)
+                }
+                return cell
+            }
+        } else {
+            cell.siteFacilityNameLabel?.text = sites[indexPath.row].name
+            cell.accessoryType = .disclosureIndicator
+                
+            // If the user clicks on Back Button, then highlight any previously selected Site.
+            if (cell.siteFacilityNameLabel.text == viewModel.selectedSite?.name) {
+                cell.siteFacilityDotImg.isHidden = false
+                cell.siteFacilityNameLabel?.font = UIFont.boldSystemFont(ofSize: 16.0)
+            }
+            return cell
+        }
+        return UITableViewCell()
+    }
+    
+    public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        if let cell = siteFacilityTableView.cellForRow(at: indexPath) as? SiteFacilityTableViewCell {
+            let sites = filteredSites()
+            let facilities = filteredFacilities()
+            
+            cell.siteFacilityDotImg.isHidden = true
+            cell.siteFacilityNameLabel?.font = UIFont.boldSystemFont(ofSize: 16.0)
+            
+            if (isShowingFacilities) {
+                viewModel.selectedFacility = facilities[indexPath.row]
+                
+                // When a facility is selected, reset the previously selected level.
+                viewModel.selectedLevel = nil
+                    
+                // Close the prompt and zoom to the selected facility.
+                closeSiteFacilityPrompt()
+                viewModel.zoomToSelection()
+                delegate?.siteFacilityIsUpdated(viewModel: viewModel)
+                
+                // Reset the search bar.
+                resetSearchFilteredResults()
+                dismissSearchBar()
+            } else {
+                viewModel.selectedSite = sites[indexPath.row]
+                selectedSiteIndexPath = indexPath
+                
+                dismissSearchBar()
+                
+                // Zoom to the map to the selected site in case the user closes the prompt without selecting a facility.
+                viewModel.zoomToSelection()
+                
+                // Reload the list to show the list of facilities for the selected site.
+                isShowingFacilities = true
+                updatePromptTitle()
+                siteFacilityTableView.reloadData()
```

Comment:
Having spent some time using the component now, I understand a bit better what is going on. This is essentially implementing a navigation interface without using a navigation controller. It even uses a disclosure indicator which indicates that when the row is tapped, a new view is pushed on the navigation stack. I'm curious why a navigation controller wasn't used? That would simplify things quite a bit and would also mean that the UI would behave the way an iOS user would expect. For example, tapping on a row should cause the new view to animate in and tapping the "back" button should cause the current view to animate out. I would strongly suggest using `UINavigationController` as the presented view controller and having separate view controllers for the facilities list and the levels list.

Reply:
Thanks for pointing that. We could look into adding a Navigation Controller as a future enhancement. I have added that in the issue of enhancements: https://devtopia.esri.com/runtime/common-toolkit/issues/166. 
@tomhillgeo